src/components/Input/Input.tsx

```diff
@@ -271,26 +125,171 @@ function renderInputAddons(
     )
   }
   if (addonAfter) {
-    return (
-      <AddonAfterSpan>
+    addonDOM = (
+      <AddonAfterSpan iconSize={getIconSize(inputSize)}>
         {children}
         {addonAfter ? (
           <AddonContainer
             css={`
-              right: 5px;
+              right: 1rem;
             `}>
             {addonAfter}
           </AddonContainer>
         ) : null}
       </AddonAfterSpan>
     )
   }
-  return children
+  return (
+    <>
+      {labelDOM}
+      {addonDOM}
+      {notificationDOM}
+    </>
+  )
+}
+
+const InputRef = props => {
+  const [currentValue, setCurrentValue] = useState(props.value || "")
+  const [isPasswordField, setIsPasswordField] = useState(
+    props.type === "password"
+  )
+  const [isOverflow, setIsOverflow] = useState(false)
+
+  const inputRef = props.ref || useRef<HTMLInputElement>()
```

Comment:
we are doing forwardReferencing here.. please reconsider, ref flow & use-case in this component again.